libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageFragment.kt

```diff
@@ -189,33 +140,27 @@ class PreviewImageFragment : Fragment() {
 
         // Set adapter data before the ViewPager2.restorePendingState gets called
         // to avoid manual handling of the ViewPager2 state restoration.
-        viewModel.uiState.value?.let {
-            (previewImageViewPager.adapter as PreviewImageAdapter).submitList(it.viewPagerItemsStates)
-            cropActionMenu?.isEnabled = it.editActionsEnabled
-            UiHelpers.updateVisibility(thumbnailsTabLayout, it.thumbnailsTabLayoutVisible)
-        }
+        viewModel.uiState.value?.let { updateUiState(it) }
     }
 
     private fun initializeInsertButton() {
-        insertButton.text = getString(string.insert_label_with_count, imageDataList.size)
+        insertButton.text = getString(string.insert_label_with_count, numberOfImages)
         insertButton.setOnClickListener {
-            Log.d("PreviewImageFragment", "Insert button clicked")
+            viewModel.onInsertClicked()
         }
     }
 
     private fun initializeViewModels(nonNullIntent: Intent) {
         viewModel = ViewModelProvider(this).get(PreviewImageViewModel::class.java)
         setupObservers()
-        // TODO: replace dummy list
-        viewModel.onCreateView(imageDataList)
+        val inputData = nonNullIntent.getParcelableArrayListExtra<EditImageData.InputData>(ARG_EDIT_IMAGE_DATA)
+        numberOfImages = inputData.size
```

Comment:
I had to change the way we retrieve the number of images. I think we might want to move the logic into the VM as it seems inconsistent now. I run out of time though.

Reply:
Moved here: cfe8cdf